### DISCUSSION TITLE
Scenario "Singara": Docker and Kubernetes web app not working.

### DISCUSSION THREAD

**Discussion by fduran** [SadServers/sadservers#41] [Scenarios]
Discussions about the scenario [Singara](https://github.com/fduran/sadservers/tree/main/scenarios/singara)

**Comment by jcanseco** [comment]
Interesting. I had a different approach from the [official solution](https://github.com/fduran/sadservers/tree/main/scenarios/singara#clues).

I ended up copying the `webapp` image from docker to k3s using:

```shell
$ sudo docker save webapp:latest | sudo k3s ctr images import -
```

And then changed the container's `imagePullPolicy` from `Always` -> `IfNotPresent`.

**Reply by fduran** [reply]
Thanks Jazel! Your solution is simpler but it depends on this k3s feature that can import an image to it without a repository, as in if this wasn't k3s but a regular k8s cluster there would be no direct equivalent (that I know of). OTOH there's no "official" solution, I love it when somebody comes with other possible solutions 😄

**Reply by jcanseco** [reply]
Oh yes, I was just sharing what I had come up with cause I was interested in seeing what other people come up with :)

It was an interesting problem. Thanks for putting all of this together!

**Reply by fduran** [reply]
thank you for your comment!

**Comment by SelfhostedPro** [comment]
Port forwarding to enable you to curl it from localhost doesn't feel like a proper solution on the networking side of things.

On another note, if you try to modify the k3s service to allow for the nodeport to bind to 8888 and restart the service, your server can kick you out and not allow you back in.

Was following this discussion from the k3s repo: https://github.com/k3s-io/k3s/discussions/8050

Maybe giving some more instruction around what networking/ingress is allowed or should be expected would be a good idea.
